portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -229,15 +227,65 @@ function allowUserSubmit() {
       });
 }
 
-/* inserts a functioning searchbar into the navigation bar of a page. */
-function insertSearch() {
+/**
+ * Gets the other html pages as documents and invokes method to insert Search
+ * bar
+ */
+async function loadSearch() {
+  const otherDocs =
+      await Promise.all([getHTML('/about.html'), getHTML('/statistics.html')]);
+
+  let otherElements = [];
+  otherDocs.forEach((doc) => {
+    otherElements = [...otherElements, ...Array.from(doc.body.childNodes)];
+  });
+
+  insertSearch(otherElements);
+}
+
+/**
+ * Get HTML asynchronously as document
+ * @param {String} url: The URL to get HTML from
+ */
+async function getHTML(url) {
+  return new Promise((resolve, reject) => {
+    // Feature detection
+    if (!window.XMLHttpRequest) return;
+
+    // Create new request
+    const xhr = new XMLHttpRequest();
```

Comment:
What is xhr? A more specific variable name would be helpful

Reply:
The request returns an XMLHttpRequest (XHR) object, thus its the most appropriate name.